Foundation/NSData.swift

```diff
@@ -453,13 +453,27 @@ open class NSData : NSObject, NSCopying, NSMutableCopying, NSSecureCoding {
         let permissions = (try? fm._permissionsOfItem(atPath: path)) ?? 0o600
 
         if writeOptionsMask.contains(.atomic) {
-            let (newFD, auxFilePath) = try _NSCreateTemporaryFile(path)
-            let fh = FileHandle(fileDescriptor: newFD, closeOnDealloc: true)
+            var (newFD, auxFilePath) = try _NSCreateTemporaryFile(path)
+#if os(Windows)
+            let shouldCleanUpFD = false
+#else
+            let shouldCleanUpFD = true
+#endif
+            let fh = FileHandle(fileDescriptor: newFD, closeOnDealloc: shouldCleanUpFD)
             do {
                 try doWrite(fh)
+#if os(Windows)
+                // Moving a file (via _NSCleanupTemporaryFile) requires that
+                // there be no open handles to the file
+                _close(newFD)
```

Comment:
I get that we need to close the handle, but I think that we can share more of the path by using `FileHandle.closeFile()` on Windows explicitly.